### PR TITLE
Encode Redirect URI

### PR DESCRIPTION
**Pull request by 0xSalman**
This is a fix for #113 

As the title says, encoded redirect url for all providers. Tested with facebook and linkedin and seems to be working well.

### CODE REVIEW

**Comment by RWOverdijk** [comment]
@msalman86 Awesome! Thanks a lot.

@doktordirk Looking at this... Perhaps we should put this in a variable somewhere? I think it's sort of off that we have so many copies of the same thing.

**Comment by doktordirk** [comment]
@RWOverdijk  https://github.com/SpoonX/aurelia-authentication/issues/119
@msalman86 can you rephrase your commit message to 
`fix(BaseConfig): bla bla ..` . better for our changelog (see contributing.md)

**Comment by 0xSalman** [comment]
@doktordirk 
Updated commit comment and also rebased

**Comment by doktordirk** [comment]
@msalman86 tx again. and merged it is :smile:
